From 84fb7d8802269eef1214f7af34e7484134ac6400 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20Men=C5=A1=C3=ADk?= <pemensik@redhat.com>
Date: Thu, 12 Jun 2025 18:16:08 +0200
Subject: [PATCH] Fix few warnings generated by gcc in tests

Ensure memory allocation is checked by the test. In vector this change
should not be required, but gcc emits warning about access to
uniinitialized src. That should never be accessed with srccnt == 0, but
add source initialization to keep compiler quiet.
---
 tests/ngtcp2_ringbuf_test.c | 8 ++++++--
 tests/ngtcp2_vec_test.c     | 2 +-
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/tests/ngtcp2_ringbuf_test.c b/tests/ngtcp2_ringbuf_test.c
index 644d1c5c..5a66560a 100644
--- a/tests/ngtcp2_ringbuf_test.c
+++ b/tests/ngtcp2_ringbuf_test.c
@@ -49,8 +49,10 @@ void test_ngtcp2_ringbuf_push_front(void) {
   ngtcp2_ringbuf rb;
   const ngtcp2_mem *mem = ngtcp2_mem_default();
   size_t i;
+  int r;
 
-  ngtcp2_ringbuf_init(&rb, 64, sizeof(ints), mem);
+  r = ngtcp2_ringbuf_init(&rb, 64, sizeof(ints), mem);
+  assert(r == 0);
 
   for (i = 0; i < 64; ++i) {
     ints *p = ngtcp2_ringbuf_push_front(&rb);
@@ -78,8 +80,10 @@ void test_ngtcp2_ringbuf_pop_front(void) {
   ngtcp2_ringbuf rb;
   const ngtcp2_mem *mem = ngtcp2_mem_default();
   size_t i;
+  int r;
 
-  ngtcp2_ringbuf_init(&rb, 4, sizeof(ints), mem);
+  r = ngtcp2_ringbuf_init(&rb, 4, sizeof(ints), mem);
+  assert(r == 0);
 
   for (i = 0; i < 5; ++i) {
     ints *p = ngtcp2_ringbuf_push_front(&rb);
diff --git a/tests/ngtcp2_vec_test.c b/tests/ngtcp2_vec_test.c
index 246ce92b..86c032f1 100644
--- a/tests/ngtcp2_vec_test.c
+++ b/tests/ngtcp2_vec_test.c
@@ -600,7 +600,7 @@ void test_ngtcp2_vec_copy_at_most(void) {
   /* 0 length vectors */
   {
     ngtcp2_vec dst[1];
-    const ngtcp2_vec src[1];
+    const ngtcp2_vec src[1] = {};
 
     n = ngtcp2_vec_copy_at_most(dst, 0, src, 0, 100);
 
-- 
2.49.0

